subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotterStatistics.java

```diff
@@ -104,39 +104,39 @@ public String toString() {
     }
 
     abstract class CollectingFileVisitor implements FileVisitor<Path> {
-        private final Collector collector;
+        protected final Collector collector;
```

Comment:
Why does this need to be `protected`?

Reply:
We create a new `PathVisitor`, and the new path visitor needs the collector. I can also make the collector a field in `PathVisitor` as well.